### ISSUE TITLE
False positive "Misaligned table fence" for empty table cells by `remark-lint-table-pipe-alignment` 

### ISSUE THREAD

**Issue by sisp** [remarkjs/remark-lint#309]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

remark-lint-table-pipe-alignment (ef0f45930a847030b00310949a8dd3ba257623a8)

### Steps to reproduce

Patch `packages/remark-lint-table-pipe-alignment/index.js` with

```diff
index 2c6f10f..fdfb3c9 100644
--- a/packages/remark-lint-table-pipe-alignment/index.js
+++ b/packages/remark-lint-table-pipe-alignment/index.js
@@ -56,12 +56,19 @@
  *   3:17-3:18: Misaligned table fence
  *
  * @example
- *   {"name": "ok-empty-cells.md", "gfm": true}
+ *   {"name": "ok-empty-columns.md", "gfm": true}
  *
  *   | | B     |   |
  *   |-| ----- | - |
  *   | | Bravo |   |
  *
+ * @example
+ *   {"name": "ok-empty-cells.md", "gfm": true}
+ *
+ *   | | B     |   |
+ *   |-| ----- | - |
+ *   |a| Bravo | c |
+ *
  */
 
 /**
```

then run:

```shell
npm run test-coverage
```

### Expected behavior

No error

### Actual behavior

```console
$ npm run test-coverage
...
    # Subtest: table-pipe-alignment
    not ok 64 - table-pipe-alignment
      ---
      duration_ms: 23.635949
      failureType: 'testCodeFailure'
      error: 'should equal with position'
      code: 'ERR_ASSERTION'
      stack: |-
        assertFixture (file:///.../remark-lint/test.js:385:10)
        TestContext.<anonymous> (file:///.../remark-lint/test.js:317:23)
        async Test.run (node:internal/test_runner/test:487:9)
        async TestContext.<anonymous> (file:///.../remark-lint/test.js:298:7)
        async Test.run (node:internal/test_runner/test:487:9)
        async Test.processPendingSubtests (node:internal/test_runner/test:252:7)
      ...
```

The underlying error is `Misaligned table fence`.

### Runtime

Node v16

### Package manager

npm 8

### OS

Linux

**Comment by ChristianMurphy** [comment]
Heya @sisp! 👋 
Thanks for reaching out!

To clarify what you are saying.
You are seeing a case where a missing table header can cause a table you would expect to ***not*** to be flagged, to instead ***be*** flagged.
Is that accurate?

Would you be open to creating a PR to add the test case, and perhaps resolve the issue?

**Comment by wooorm** [comment]
Fixed!

**Comment by sisp** [comment]
Perfect, thanks @wooorm! :pray:

Sorry for not responding earlier, @ChristianMurphy. I expect the case where only some cells are empty but not entire columns to pass the linter rule. @wooorm fixed it already. :+1:
